beacon-chain/operations/slashings/service.go

```diff
@@ -0,0 +1,207 @@
+package slashings
+
+import (
+	"context"
+	"sort"
+	"sync"
+
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	beaconstate "github.com/prysmaticlabs/prysm/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"github.com/prysmaticlabs/prysm/shared/sliceutil"
+)
+
+// Pool implements a struct to maintain pending and recently included voluntary exits. This pool
+// is used by proposers to insert into new blocks.
+type Pool struct {
+	lock                    sync.RWMutex
+	pendingProposerSlashing []*ethpb.ProposerSlashing
+	pendingAttesterSlashing []*PendingAttesterSlashing
+	included                map[uint64]bool
+}
+
+// PendingAttesterSlashing represents an attester slashing in the operation pool.
+// Allows for easy binary searching of included validator indexes.
+type PendingAttesterSlashing struct {
```

Comment:
How come we dont do the same for proposer?

Reply:
Every proposal slashing only has one validator index to slash, not multiple like a attestation slashing can have. So we wrap it in order to easily sort and binary search them.